plugin/documents.py

```diff
@@ -529,80 +515,19 @@ def _on_color_boxes(self, response: Any) -> None:
             for sv in self.session_views_async():
                 if sv.session == session:
-                    for request_id, request in sv.active_requests.items():
-                        if request.method == "codeAction/resolve":
-                            session.send_notification(Notification("$/cancelRequest", {"id": request_id}))
-            name = session.config.name
-            session.send_request_async(
-                Request("textDocument/codeLens", params, self.view),
-                lambda r: self._on_code_lenses_async(name, r))
-
-    def _on_code_lenses_async(self, name: str, response: Optional[List[CodeLens]]) -> None:
-        for i in range(0, len(self._code_lenses)):
-            self.view.erase_regions(self._code_lens_key(i))
-        self._code_lenses.clear()
-        if not isinstance(response, list):
-            return
-        for index, c in enumerate(response):
-            region = range_to_region(Range.from_lsp(c["range"]), self.view)
-            self._code_lenses.append((c, region))
-            if "command" in c:
-                # We consider a code lens that has a command to be already resolved.
-                self._on_resolved_code_lens_async(name, index, region, c)
-            else:
-                self._render_code_lens(name, index, region, None)
-        self._code_lenses = list((c, range_to_region(Range.from_lsp(c["range"]), self.view)) for c in response)
-        self._resolve_visible_code_lenses_async()
+                    sv.start_code_lenses_async()
 
     def _resolve_visible_code_lenses_async(self) -> None:
         session = self.session_async("codeLensProvider")
-        if session:
-            for index, code_lens, region in self._unresolved_code_lenses(self.view.visible_region()):
-                callback = functools.partial(self._on_resolved_code_lens_async, session.config.name, index, region)
-                session.send_request_async(Request("codeLens/resolve", code_lens, self.view), callback)
-
-    def _on_resolved_code_lens_async(self, name: str, index: int, region: sublime.Region, code_lens: CodeLens) -> None:
-        code_lens["session_name"] = name
-        try:
-            self._code_lenses[index] = (code_lens, region)
-        except IndexError:
-            return
-        self._render_code_lens(name, index, region, code_lens["command"])
-
-    def _unresolved_code_lenses(
-        self,
-        visible: sublime.Region
-    ) -> Generator[Tuple[int, CodeLens, sublime.Region], None, None]:
-        for index, tup in enumerate(self._code_lenses):
-            code_lens, region = tup
-            if not code_lens.get("command") and visible.intersects(region):
-                yield index, code_lens, region
+        if session and session.uses_plugin():
+            for sv in self.session_views_async():
+                if sv.session == session:
+                    sv.resolve_visible_code_lenses_async()
```

Comment:
resolve for _all_ sessions attached to the view because the data is stored per session now?